packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java

```diff
@@ -235,24 +236,33 @@ public String getPatternCatalog() throws Exception {
             var modelCatalog = (EipModel) api.model(Kind.eip, name);
             catalogMap.put(modelCatalog.getJavaType(), modelCatalog);
         }
+
         for (var entry : processors.entrySet()) {
+            var sortedSchemaProperties = jsonMapper.createObjectNode();
             var processorFQCN = entry.getKey();
             var processorSchema = entry.getValue();
             var processorCatalog = catalogMap.get(processorFQCN);
-            for (var property : processorSchema.withObject("/properties").properties()) {
-                var propertyName = property.getKey();
-                var propertySchema = (ObjectNode) property.getValue();
+
+            var camelYamlDslProperties = processorSchema.withObject("/properties").properties().stream().map(Map.Entry::getKey).sorted(
+                    new CamelYamlDSLKeysComparator(processorCatalog.getOptions())
+            ).toList();
```

Comment:
@igarashitm , I ended up doing it like this because as we saw, the Catalog has more properties than the Camel YAML DSL schema, so it added the extra `expression` field for `setHeaders`.

I think, for the time being, we could move forward with this approach:
1. Get the Camel YAML DSL schema's properties
2. Sort them by using the index from the corresponding Catalog option
3. If there's no correspondence between the Camel YAML DSL and the Catalog option, we move that property to the end of the list.

Thinking out loud, I think we could use the extra `expression` field and remove the other parsing, this way we could handle all expressions the same. To be investigated.